### PR TITLE
Initial subject definition XML for ELA and Math.

### PR DESCRIPTION
**Pull request by wtritch**
Initial subject definition XSD.

This PR contains the initial ELA and Math subject definitions for the SBAC Reporting Data Warehouse.
Additional subjects must be provided in the same format for integration with the RDW system.

The XSD schema definition is probably the best place to start the review process.  It has comments defining each of the elements/attributes.

### CODE REVIEW

File: subject/RDW_Subject.xsd [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        </xs:extension>
+      </xs:simpleContent>
+    </xs:complexType>
+  </xs:element>
+
+  <!--
+    AssessmentType elements define the behavior of assessments for this subject
+    within the context of a specific assessment type.
+    A subject may have up to 3 assessment types defined, one each for
+    summative, ICA, and IAB.
```
Comment (malaffoon) [inline subject/RDW_Subject.xsd]: perhaps `"sum", "ica", "iab"` to be consistent?

File: subject/RDW_Subject.xsd [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+    <xs:complexType>
+      <xs:sequence>
+        <!-- The performance levels available for assessments of this type -->
+        <xs:element ref="PerformanceLevels"/>
+        <!-- The item difficulty cutpoints of items for assessments of this type -->
+        <xs:element ref="ItemDifficulties" minOccurs="0"/>
+        <!-- Subscore definitions for assessments of this type -->
+        <xs:element ref="SubScoring" minOccurs="0"/>
+      </xs:sequence>
+      <!-- The assessment type being defined. Must be "ica," "iab," or "sum." -->
```
Comment (malaffoon) [inline subject/RDW_Subject.xsd]: I think the victorian punctuation is confusing, perhaps just `Must be "ica", "iab" or "sum".`

File: subject/RDW_Subject.xsd [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+
+  <!-- DepthOfKnowledge depth must be a positive value greater than or equal to 1 -->
+  <xs:simpleType name="KnowledgeDepth">
+    <xs:restriction base="xs:integer">
+      <xs:minInclusive value="1"/>
+    </xs:restriction>
+  </xs:simpleType>
+
+  <!-- The assessment grades available within the RDW system -->
+  <xs:simpleType name="GradeCode">
```
[the remark this answers is not in the document]
Comment (wtritch) [reply subject/RDW_Subject.xsd]: good call, done

File: subject/RDW_Subject.xsd (hunk 2 above)
Comment (wtritch) [reply subject/RDW_Subject.xsd]: done

File: subject/RDW_Subject.xsd (hunk 1 above)
Comment (wtritch) [reply subject/RDW_Subject.xsd]: done

**Comment by jtreuting** [comment]
Thanks @alexdean201 

- Honestly, I think it will take some good effort for a vendor to put together an XML file representing a subject, but it should be a one time thing not something that is done each year since it sets rules for the subject and not the individual assessments.
- The relationship is through the subject code that is provided in the administration package (and for new subjects that don't use the SB package format it would be the code included in the CSV file used to load an assessment).
- We are still working this out and will provide more details as we get clarity on that.
- These subject files are loaded into the system and will populate many metadata lookup tables that are currently hard-coded rules for SB Math and ELA.  The reporting system will use things like the performance levels, claims and targets to generate the proper reports and label those things appropriately.  The TRT files that send scores into the system for claims and targets will need to match the codes defined by the subject in these files.  That is how the scores will get correlated to the appropriate place.

File: subject/ELA_subject.xml [hunk 4]
```diff
@@ -0,0 +1,3824 @@
...
+         xsi:schemaLocation="http://www.smarterbalanced.org/RDW_Subject RDW_Subject.xsd"
+         code="ELA"
+         name="ELA">
+  <Messages>
+    <Message key="standard-description"><![CDATA[This column reflects the Common Core State Standard(s) associated with a test item. Test items may be associated with more than one standard.  For example, a question that asks students to evaluate research sources may assess both reading and research skills.  For more information, see the <a href=\"https://portal.smarterbalanced.org/library/en/english-language-artsliteracy-content-specifications.pdf\" target=\"_blank\">English Language Arts Content Specifications</a>.]]></Message>
+  </Messages>
+  <AssessmentTypes>
+    <AssessmentType code="ica">
+      <PerformanceLevels standardCutoff="3">
+        <PerformanceLevel level="1" name="Did Not Meet Standard" shortName="Did Not Meet" suffix="Standard" color="maroon"/>
```
Comment (wtritch) [inline subject/ELA_subject.xml]: @jtreuting @esotrope I added the "shortName" and "suffix" optional display text attributes here to account for our UI displays that style "Did Not Meet" differently than "Standard" when displayed as a table-header.

File: subject/RDW_Subject.xsd [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+        attribute when visual space is a concern.  It may be displayed along with the "suffix" attribute value.
+        It may be visually styled differently than the suffix.
+        This value will default to the "name" value if undefined.
+       -->
+      <xs:attribute name="shortName" type="xs:string"/>
+      <!--
+        The suffix display string for the performance level.  This value may be displayed after the "shortName"
+        attribute and may be visually styled differently than the "shortName" text.
+      -->
+      <xs:attribute name="suffix" type="xs:string"/>
```
Comment (wtritch) [inline subject/RDW_Subject.xsd]: @jtreuting @esotrope I added some documentation to the XSD calling out how these values are used.

**Comment by wtritch** [comment]
Closing this PR for clarity

**Pull request closed without merging**